collectors/test-results/jira-xray/src/main/java/com/capitalone/dashboard/core/client/testexecution/TestExecutionClientImpl.java

```diff
@@ -0,0 +1,418 @@
+                    int skipSteps = stepCountByStatus.get("SKIPSTEP_COUNT");
+                    int unknownSteps = stepCountByStatus.get("UNKNOWNSTEP_COUNT");
+                    testCase.setTotalTestStepCount(totalSteps);
+                    testCase.setFailedTestStepCount(failSteps);
+                    testCase.setSuccessTestStepCount(passSteps);
+                    testCase.setSkippedTestStepCount(skipSteps);
+                    testCase.setUnknownStatusCount(unknownSteps);
+                    if(failSteps > 0) {
+                        testCase.setStatus(TestCaseStatus.Failure);
+                    } else if (skipSteps > 0){
+                        testCase.setStatus(TestCaseStatus.Skipped);
+                    } else if(passSteps > 0){
+                        testCase.setStatus(TestCaseStatus.Success);
+                    } else {
+                        testCase.setStatus(TestCaseStatus.Unknown);
+                    }
+
+                    testCase.setTestSteps(this.getTestSteps(testRun));
+                }
+
+            } catch (Exception e) {
+                LOGGER.error("Unable to get the Test Step: " + e);
+            }
+            testCases.add(testCase);
+        }
+
+        return testCases;
+    }
+
+    /**
+     * Gets the test steps for a test case
+     *
+     * @param testRun
+     * @return
+     */
+    private List<TestCaseStep> getTestSteps(TestRun testRun) {
+
+        List<TestCaseStep> testSteps = new ArrayList<>();
+
+        for (TestStep testStep : testRun.getSteps()) {
+            TestCaseStep testCaseStep = new TestCaseStep();
+
+            testCaseStep.setId(testStep.getId().toString());
+            testCaseStep.setDescription(testStep.getStep().getRaw());
+            if (testStep.getStatus().toString().equals("PASS")) {
+                testCaseStep.setStatus(TestCaseStatus.Success);
+            } else if (testStep.getStatus().toString().equals("FAIL")) {
+                testCaseStep.setStatus(TestCaseStatus.Failure);
+            } else {
+                testCaseStep.setStatus(TestCaseStatus.Skipped);
+            }
+            testSteps.add(testCaseStep);
+        }
+
+        return testSteps;
+    }
+
+
+    /**
+     * Gets the test cases count map based on the status {pass, fail, skip & unknown}
+     *
+     * @param testExec
+     * @param tests
+     * @return
+     */
+    private Map<String,Integer> getTestCountStatusMap(Feature testExec, Iterable<TestExecution.Test> tests) {
+
+        Map<String,Integer> map = new HashMap<String,Integer>(4);
+        int failCount = 0;
+        int passCount = 0;
+        int skipCount = 0;
+        int unknownCount = 0;
+
+        for (TestExecution.Test test : tests) {
+            try {
+                TestRun testRun = restClient.getTestRunClient().getTestRun(testExec.getsNumber(), test.getKey()).claim();
+                if (testRun != null) {
+                    if (testRun.getStatus().toString().equals("FAIL")) {
```

Comment:
@kumarvarun1252  refactor if else and extract to method and reuse

Reply:
method is refactored to have the rest call made once for all